### ISSUE TITLE
Improve dev package finder

### ISSUE THREAD

**Issue by tangkong** [pcdshub/hutch-python#332]
## Expected Behavior
get_standard_dev_pkgs() should return more reasonable list of packages

## Current Behavior
```python 
In [12]: os.environ['PYTHONPATH'] = '/cds/group/pcds/pyps/apps/hutch-python/xpp:/cds/group/pcds/pyps/apps/hutch-python/xpp/dev/devpath:/cds/group/pcds/pyps/apps/hutch-python/xpp/../common/dev
    ...: /devpath:/reg/g/pcds/dist/pds/xpp/current/build/pdsapp/lib/x86_64-rhel7-opt:/reg/g/pcds/dist/pds/xpp/ami-current/build/ami/lib/x86_64-rhel7-opt'

In [13]: get_standard_dev_pkgs()
Out[13]: {'dev', 'hutch-python'}
```

I believe the correct answer here is pcdsdevices, hutch-python, archapp but I'm not 100% on that

## Possible Solution
Probably need to expand the paths similar to how python does.  Naively, one could stop recursively adding directories once an `__init__.py` is found.  But maybe there's a better way to do this.  

## Context
#330 

## Your Environment
pcds-5.3.1

**Comment by ZLLentz** [comment]
I think looking for `__init__.py` was my first thought as well. It's a bit annoying but it'll make the output more thorough.

I think we can just:
- check every directory in the pythonpath (at depth 1)
- if there's an `__init__.py`, that's a module to include
- if there's not such a file, skip it

I guess the main things this misses are compiled c python modules and random .py files? I suppose we can include those too

**Comment by tangkong** [comment]
I would have voted for `setup.py` at first but some of the "packages" were flattened past that point.

I think your suggested procedure should work though

**Comment by klauer** [comment]
I believe the correct/most accurate method here might be to use `pkgutil` - but it's relatively slow. 
"Correct" isn't necessarily right for us, though. I think the `__init__.py` method will likely suffice for what we're after.

<details>

<summary>For reference</summary>

`iter_modules` shows my dev packages pretty readily:

```python
In [26]: pkgutil.iter_modules?
Signature: pkgutil.iter_modules(path=None, prefix='')
Docstring:
Yields ModuleInfo for all submodules on path,
or, if path is None, all top-level modules on sys.path.

'path' should be either None or a list of paths to look for
modules in.

'prefix' is a string to output on the front of every module name
on output.
File:      ~/miniforge3/lib/python3.8/pkgutil.py
Type:      function

In [27]: list(pkgutil.iter_modules())
Out[27]:
[ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='ads_async', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='beautysh', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='happi', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='hutch_python', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='lucid', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='ophyd', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='pcdsdevices', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='pcdsutils', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='pydm', ispkg=True),
 ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='typhos', ispkg=True),
...
```

`walk_packages` does things recursively, which is undesirable here:

```python
In [8]: pkgutil.walk_packages?
Signature: pkgutil.walk_packages(path=None, prefix='', onerror=None)
Docstring:
Yields ModuleInfo for all modules recursively
on path, or, if path is None, all accessible modules.

'path' should be either None or a list of paths to look for
modules in.

'prefix' is a string to output on the front of every module name
on output.

Note that this function must import all *packages* (NOT all
modules!) on the given path, in order to access the __path__
attribute to find submodules.

'onerror' is a function which gets called with one argument (the
name of the package which was being imported) if any exception
occurs while trying to import a package.  If no onerror function is
supplied, ImportErrors are caught and ignored, while all other
exceptions are propagated, terminating the search.

Examples:

# list all modules python can access
walk_packages()

# list all submodules of ctypes
walk_packages(ctypes.__path__, ctypes.__name__+'.')
File:      ~/miniforge3/lib/python3.8/pkgutil.py
Type:      function

In [9]: list(pkgutil.walk_packages(path=["/cds/home/k/klauer/dev_path"]))
[ModuleInfo(module_finder=FileFinder('/cds/home/k/klauer/dev_path'), name='ads_async', ispkg=True),
...
```
</details>

**Comment by tangkong** [comment]
Closed by #333
